### PR TITLE
Add routes and controllers for team-endpoints

### PR DESCRIPTION
**Pull request by carpecodeum**
# Description

Add routes and controllers for team-endpoints.

Fixes #424 

Annotation 2020-06-30 162442
Annotation 2020-06-30 163041

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by carpecodeum** [comment]
@ivantha @Tushar19varshney @UdeshUK please review this :)

**Comment by hiranthaR** [comment]
@carpecodeum please resolve conflicts and change status codes to right one in teamController :slightly_smiling_face:

File: labellab-flask/api/controllers/teamscontroller.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+            return make_response(jsonify(response)), 200
+
+        except Exception as err:
+            print("Error occured: ", err)
+            response = {
+                "success": False,
+                "msg": "Data could not be fetched."
+                }
+            return make_response(jsonify(response)), 404
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:69]: better to send 500,

File: labellab-flask/api/controllers/teamscontroller.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+    @jwt_required
+    def get(self, project_id, team_id):
+        """Handle GET request for this view. Url --> /api/v1/team/team_info/<int:project_id>/<int:team_id>"""
+        try:
+            if not team_id:
+                response = {
+                    "success": False,
+                    "msg": "Provide all the required data."
+                }
+                return make_response(jsonify(response)), 500
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:82]: if team id not valid 400

File: labellab-flask/api/controllers/teamscontroller.py [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+                "msg": "Team found.",
+                "body": team
+            }
+            return make_response(jsonify(response)), 200
+        except Exception:
+            response = {
+                "success": False,
+                "msg": "Something went wrong!!"
+            }
+            return make_response(jsonify(response)), 404
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:102]: better to send 500

**Comment by carpecodeum** [comment]
> @carpecodeum please resolve conflicts and change status codes to right one in teamController 🙂

Done! 🙂

**Comment by ivantha** [comment]
@hiranthaR Are the issues resolved? Can we merge this?

File: labellab-flask/api/controllers/teamscontroller.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+        current_user = get_jwt_identity
+        roles = get_user_roles(current_user, project_id)
+        
+        if not roles.index("admin"):
+            print("Error occured: user not admin")
+            response = {
+                    "success": False,
+                    "msg": "User not admin."
+                }
+            return make_response(jsonify(response)), 400
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:116]: User is not the admin. so he cannot delete it. so status should should be 403 right?

File: labellab-flask/api/controllers/teamscontroller.py [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+            response = {
+                "success": False,
+                "msg": "Something went wrong!!"
+            }
+            return make_response(jsonify(response)), 500
+    
+    @jwt_required
+    def delete(self, project_id, team_id):
+        """Handle DELETE request for this view. Url --> /api/v1/team/team_info/<int:project_id>/<int:team_id>"""
+        current_user = get_jwt_identity
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:107]: get_jwt_identity() missed () these.
did you run the code??

File: labellab-flask/api/controllers/teamscontroller.py [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+                    "msg": "User not admin."
+                }
+            return make_response(jsonify(response)), 400
+        try:
+            if not team_id:
+                response = {
+                    "success": False,
+                    "msg": "Provide all the required data."
+                }
+                return make_response(jsonify(response)), 500
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:123]: team Id is missing. bad request 400

File: labellab-flask/api/controllers/teamscontroller.py [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+            }
+            return make_response(jsonify(response)), 500
+
+    @jwt_required
+    def put(self, project_id, team_id):
+        """Handle PUT request for this view. Url --> /api/v1/team/team_info/<int:project_id>/<int:team_id>"""
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,
+                                     force=True)
+        current_user = get_jwt_identity
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:143]: get_jwt_identity() ??

File: labellab-flask/api/controllers/teamscontroller.py [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+        roles = get_user_roles(current_user, project_id)
+        
+        if not roles.index("admin"):
+            print("Error occured: user not admin")
+            response = {
+                    "success": False,
+                    "msg": "User not admin."
+                }
+            return make_response(jsonify(response)), 400
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:152]: User is not allowed to update 403

File: labellab-flask/api/controllers/teamscontroller.py [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+            }
+            return make_response(jsonify(response)), 400
+        
+        try:
+            if not (team_id):
+                response = {
+                    "success": False,
+                    "msg": "Provide all the required data."
+                }
+                return make_response(jsonify(response)), 500
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:169]: team_id not provided so bad request 400

File: labellab-flask/api/controllers/teamscontroller.py [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+    @jwt_required
+    def post(self, project_id, team_id):
+        """
+        Handle POST request for this view.
+        Url --> /api/v1/team/add_team_member/<int:project_id>/<int:team_id>
+        """
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,
+                                     force=True)
+        current_user = get_jwt_identity
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:213]: get_jwt_identity()

File: labellab-flask/api/controllers/teamscontroller.py [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+        roles = get_user_roles(current_user, project_id)
+
+        if not roles.index("admin"):
+            print("Error occured: user not admin")
+            response = {
+                    "success": False,
+                    "msg": "User not admin."
+                }
+            return make_response(jsonify(response)), 400
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:222]: 403

File: labellab-flask/api/controllers/teamscontroller.py [hunk 12]
```diff
@@ -0,0 +1,329 @@
...
+    @jwt_required
+    def post(self, project_id, team_id):
+        """
+        Handle POST request for this view.
+        Url --> /api/v1/team/remove_team_member/<int:project_id>/<int:team_id>
+        """
+        # getting JSON data from request
+        post_data = request.get_json(silent=True,
+                                     force=True)
+        current_user = get_jwt_identity
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:278]: get_jwt_identity()

File: labellab-flask/api/controllers/teamscontroller.py [hunk 13]
```diff
@@ -0,0 +1,329 @@
...
+        roles = get_user_roles(current_user, project_id)
+
+        if not roles.index("admin"):
+            print("Error occured: user not admin")
+            response = {
+                    "success": False,
+                    "msg": "User not admin."
+                }
+            return make_response(jsonify(response)), 400
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:287]: 403

File: labellab-flask/api/controllers/teamscontroller.py [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+                response = {
+                    "success": False,
+                    "msg": "Could not delete projectmember from all teams"}
+                return make_response(jsonify(response)), 400
+
+        except Exception:
+            response = {
+                "success": False,
+                "msg": "Please provide all the required fields."}
+            return make_response(jsonify(response)), 404
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:321]: 400

File: labellab-flask/api/controllers/teamscontroller.py [hunk 15]
```diff
@@ -0,0 +1,329 @@
...
+                    "success": True,
+                    "msg": "Team member deleted."
+                }
+                return make_response(jsonify(response)), 200
+
+            except Exception:
+                response = {
+                    "success": False,
+                    "msg": "Could not delete projectmember from all teams"}
+                return make_response(jsonify(response)), 400
```
Comment (hiranthaR) [inline labellab-flask/api/controllers/teamscontroller.py:315]: 500

File: labellab-flask/api/controllers/teamscontroller.py (hunk 4 above)
Comment (carpecodeum) [reply labellab-flask/api/controllers/teamscontroller.py:116]: > Fix the status codes then run and check your code, please. 🙂

Sorry these errors may have arise when resolving conflicts. Will fix them right away !

**Comment by carpecodeum** [comment]
> Fix the status codes then run and check your code, please. 🙂

Sorry issues may have come up due to merge conflicts resolving. Issues are fixed now.

**Pull request merged**